R/as_curve_data.R

```diff
@@ -0,0 +1,60 @@
+#' Load antibody decay curve parameter
+#'
+#' @param data a [data.frame()] or [tibble::tbl_df]
+#' @param antigen_isos [character()] vector of antigen isotypes to be used in analyses
+#' @returns a `curve_data` object (a [tibble::tbl_df] with extra attribute `antigen_isos`)
+#' @export
+#' @examples
+#' curve_data <- load_curve_params("https://osf.io/download/rtw5k/")
+#'
+#' print(curve_data)
+#'
+as_curve_data <- function(data, antigen_isos = NULL) {
+
+  # convert data to a tibble data type
+  tibble_data <-
+    data %>%
+    tibble::as_tibble()
```

Comment:
none of this code is currently tested; easiest fix will be to call it in `load_curve_params()`, then tests of that function will implicitly test this function as well.